### PR TITLE
Add changes for crdsdev

### LINKED ISSUE #69: CAPOCI should publish CRD reference
**What would you like to be documented**:
CAPOCI should publish CRD reference

**Why is this needed**:
This will help users to understand the CRD, which the cluster templates are based on. Any customisation to the template will be greatly helped by this documentation.

### PR DESCRIPTION
**Pull request by shyamradhakrishnan**
**What this PR does / why we need it**:
The change will help publish API Reference docs here - https://doc.crds.dev/ . See this as an example - https://doc.crds.dev/github.com/shyamradhakrishnan/cluster-api-provider-oci@v0.5.0-alpha5

Removing the vendor folder as that is the best practise across all go projects and help us remove unnecessary CRSs in the docs.

Renaming the CNI yamls so that the CRDS are not published.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #69

### CODE REVIEW

**Comment by shyamradhakrishnan** [comment]
e2e tests
```
• [SLOW TEST:743.154 seconds]
Workload cluster creation
/home/ubuntu/go/src/github.com/oracle/cluster-api-provider-oci/test/e2e/cluster_test.go:50
  ClusterClass - with 1 control-plane nodes and 1 worker nodes [PRBlocking]
  /home/ubuntu/go/src/github.com/oracle/cluster-api-provider-oci/test/e2e/cluster_test.go:471
------------------------------
STEP: Tearing down the management cluster


Ran 5 of 22 Specs in 1774.223 seconds
SUCCESS! -- 5 Passed | 0 Failed | 0 Pending | 17 Skipped


Ginkgo ran 1 suite in 30m25.172649717s
Test Suite Passed
make[1]: Leaving directory '/home/ubuntu/go/src/github.com/oracle/cluster-api-provider-oci'
``

**Comment by shyamradhakrishnan** [comment]
unit tests
```
?   	github.com/oracle/cluster-api-provider-oci	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/api/v1beta1	1.040s	coverage: 17.9% of statements
?   	github.com/oracle/cluster-api-provider-oci/cloud/config	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/cloud/ociutil	0.823s	coverage: 25.0% of statements
ok  	github.com/oracle/cluster-api-provider-oci/cloud/scope	167.680s	coverage: 73.2% of statements
?   	github.com/oracle/cluster-api-provider-oci/cloud/scope/mocks	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/base	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/base/mock_base	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/compute	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/compute/mock_compute	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/computemanagement	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/containerengine	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/containerengine/mock_containerengine	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/identity	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/identity/mock_identity	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer/mock_nlb	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/vcn	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/vcn/mock_vcn	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/controllers	28.444s	coverage: 69.0% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/api/v1beta1	2.564s	coverage: 10.1% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/controllers	1.654s	coverage: 43.4% of statements
?   	github.com/oracle/cluster-api-provider-oci/feature	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/version	[no test files]
```

File: test/e2e/data/cni/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+The yaml file extension for the files in this folder has been removed so that the CRDs referenced
+in the files are not added to CAPOCI API 
+reference documentation here https://doc.crds.dev/github.com/oracle/cluster-api-provider-oci
```
Comment (joekr) [inline test/e2e/data/cni/README.md:3]: awesome callout! Thanks for this 😄

File: Dockerfile [hunk 2]
```diff
@@ -17,7 +17,6 @@ COPY controllers/ controllers/
 COPY cloud/ cloud/
 COPY exp/ exp/
 COPY feature/ feature/
-COPY vendor/ vendor/
```
Comment (joekr) [inline Dockerfile:20]: Is this no longer needed as part of the image build?

**Review by joekr — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (shyamradhakrishnan) [reply Dockerfile:20]: yes, because the vendoring will be don by the build command, this was needed if we needed to mount the vendor

**Pull request merged**